src/_pytest/logging.py

```diff
@@ -402,6 +402,17 @@ def get_records(self, when: str) -> List[logging.LogRecord]:
         """
         return self._item.stash[caplog_records_key].get(when, [])
 
+    def get_records_by_logger(self, logger: str) -> List[logging.LogRecord]:
```

Comment:
What about adding an optional logger parameter to get_records instead of introducing a new method in the LogCaptureFixture class?